sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestDefineInParameter.java

```diff
@@ -0,0 +1,154 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDefineInParameter
+{
+    @Rule
+    public H2DatabaseRule db = new H2DatabaseRule().withPlugin(new SqlObjectPlugin());
+
+    private Handle handle;
+    private List<String> testColumns;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        handle = db.getSharedHandle();
+        handle.execute("create table test (id identity primary key, name varchar(50))");
+        handle.execute("create table testNullable (id identity primary key, name varchar(50) null)");
+        testColumns = new ArrayList<>();
+        testColumns.add("id");
+        testColumns.add("name");
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        handle.execute("drop table test");
+        handle.execute("drop table testNullable");
+        handle.close();
+    }
+
+    @Test
+    public void testWithBindIn() throws Exception
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+
+        List<Object> values = new ArrayList<>();
+        values.add(1);
+        values.add("Some Pig");
+
+        List<Object> valuesNull = new ArrayList<>();
+        valuesNull.add(2);
+        valuesNull.add(null);
+
+        testDao.insert("test", testColumns, values);
+        testDao.insert("testNullable", testColumns, valuesNull);
+
+        Something something = new Something(1, "Some Pig");
+        Something nothing = new Something(2, null);
+
+        assertThat(testDao.findById(testColumns, "test", 1)).isEqualTo(something);
+        assertThat(testDao.findById(testColumns, "testNullable", 1)).isNull();
+
+        assertThat(testDao.findById(testColumns, "test", 2)).isNull();
+        assertThat(testDao.findById(testColumns, "testNullable", 2)).isEqualTo(nothing);
+    }
+
+    @Test
+    public void testWithBindBean() throws Exception
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+
+        Something something = new Something(1, "Some Pig");
+        Something nothing = new Something(2, null);
+
+        testDao.insert("test", testColumns, something);
+        testDao.insert("testNullable", testColumns, nothing);
+
+        assertThat(testDao.findById(testColumns, "test", 1)).isEqualTo(something);
+        assertThat(testDao.findById(testColumns, "testNullable", 1)).isNull();
+
+        assertThat(testDao.findById(testColumns, "test", 2)).isNull();
+        assertThat(testDao.findById(testColumns, "testNullable", 2)).isEqualTo(nothing);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIllegalArray()
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+        String[] columnsArray = {"id", "name"};
+        List<Object> values = new ArrayList<>();
+        values.add(1);
+        values.add("Some Pig");
+        testDao.badInsert("test", columnsArray, values);
+    }
```

Comment:
Actually, I'd like array support. We just added it to `@BindIn` I want to say last week.